backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/search/predicate/impl/AbstractElasticsearchSearchPredicateBuilder.java

```diff
@@ -6,22 +6,36 @@
  */
 package org.hibernate.search.backend.elasticsearch.search.predicate.impl;
 
+import java.util.LinkedList;
+import java.util.List;
+
 import org.hibernate.search.backend.elasticsearch.gson.impl.JsonAccessor;
 import org.hibernate.search.engine.search.predicate.spi.SearchPredicateBuilder;
 
 import com.google.gson.JsonObject;
 
 
-
 public abstract class AbstractElasticsearchSearchPredicateBuilder
```

Comment:
Some predicates, such as the `id` and `bool` predicate, have nothing to do with fields.

I believe it would make more sense to have a dedicated abstract class for predicates that support nested. Just like you did with `AbstractElasticsearchSearchNestedSortBuilder`, which extends `AbstractElasticsearchSearchSortBuilder` and adds nesting support.